specification/communication/Communication.Messages/models.tsp

```diff
@@ -0,0 +1,370 @@
+  @doc("Text message type.")
+  text,
+
+  @doc("Image message type.")
+  image,
+
+  @doc("Template message type.")
+  template,
+}
+
+@access(Access.public)
+@doc("The WhatsApp button sub type.")
+enum WhatsAppMessageButtonSubType {
+  @doc("The WhatsApp button sub type is quick reply.")
+  quickReply,
+
+  @doc("The WhatsApp button sub type is url.")
+  url,
+}
+
+@access(Access.public)
+@doc("The aggregated template status.")
+enum MessageTemplateStatus {
+  @doc("Message template is approved.")
+  approved,
+
+  @doc("Message template is rejected.")
+  rejected,
+
+  @doc("Message template is pending.")
+  pending,
+
+  @doc("Message template is paused.")
+  paused,
+}
+
+@doc("The type of the template parameter.")
+enum MessageTemplateValueType {
+  @doc("The text template parameter type.")
+  text,
+
+  @doc("The image template parameter type.")
+  image,
+
+  @doc("The document template parameter type.")
+  document,
+
+  @doc("The video template parameter type.")
+  video,
+
+  @doc("The location template parameter type.")
+  location,
+
+  @doc("The quick action template parameter type.")
+  quickAction,
+}
+
+@doc("The type of the communication messages channel.")
+enum CommunicationMessagesChannel {
+  @doc("The WhatsApp communication messages channel type.")
+  whatsApp,
+}
+
+@doc("The type of the message template.")
+enum MessageTemplateBindingsType {
+  @doc("The WhatsApp template type.")
+  whatsApp,
+}
+
+@discriminator("kind")
+@doc("Details of the message to send.")
+model NotificationContent {
+  @doc("The Channel Registration ID for the Business Identifier.")
+  channelRegistrationId: uuid;
+
+  @doc("The native external platform user identifiers of the recipient.")
+  to: string[];
+
+  @doc("The type discriminator describing a notification type.")
+  kind: CommunicationMessageType;
+}
+
+@doc("A request to send a text notification.")
+model TextNotificationContent extends NotificationContent {
+  @doc("Message notification type is text.")
+  kind: CommunicationMessageType.text;
+
+  @doc("Message content.")
+  content: string;
+}
+
+@doc("A request to send a media notification.")
+model MediaNotificationContent extends NotificationContent {
+  @doc("Message notification type is image.")
+  kind: CommunicationMessageType.image;
+
+  @doc("Optional text content.")
+  content?: string;
+
+  @doc("""
+  A media url for the file. Required if the type is one of the supported media
+  types, e.g. image
+  """)
+  mediaUri: url;
+}
+
+@doc("A request to send a template notification.")
+model TemplateNotificationContent extends NotificationContent {
+  @doc("Message notification type is template.")
+  kind: CommunicationMessageType.template;
+
+  @doc("The template object used to create templates.")
+  template: MessageTemplate;
+}
+
+@doc("The template object used to create templates.")
+model MessageTemplate {
+  @doc("Name of the template.")
+  name: string;
+
+  @doc("The codes for the supported languages for templates.")
```

Comment:
In what format? Other services specify the spec e.g. BCP47 e.g., "en", "en-US", etc. Specify in you doc string.

Reply:
Added the format and more description